cylc/flow/tests/test_rundb.py

```diff
@@ -53,11 +57,117 @@ def test_select_task_job_sqlite_error(self):
         """Test that when the rundb CylcSuiteDAO select_task_job method raises
         a SQLite exception, the method returns None"""
 
-        self.mocked_connection.execute.side_effect = DatabaseError
+        self.mocked_connection.execute.side_effect = sqlite3.DatabaseError
 
         r = self.dao.select_task_job("it'll", "raise", "an error!")
         self.assertIsNone(r)
 
 
+def create_temp_db():
+    temp_db = mktemp()
+    conn = sqlite3.connect(temp_db)
```

Comment:
For testing, can use `':memory:'` to create a database in RAM.

Reply:
Need to save to disk so the DB can be re-opened by the Cylc DAO.

Reply:
Cam that be mocked?

Reply:
If you know how...

Reply:
Otherwise, please make sure that the temporary file is cleared out after the test.

Reply:
Not sure how to mock, easy to housekeep.

Reply:
nice

Reply:
I think a temporary file can be used. @dwsutherland 's PR has some new test utility code that handles something similar (though I haven't followed the whole discussion here, so forgive me if out of context): https://github.com/cylc/cylc-flow/pull/3202/files#diff-8222eb14a16268057dbc4ac79670fa3fR109-R113

(in David's PR, there is a base class for tests, so the suite DB is created on `setUp`, and destroyed on `tearDown`)